tokio/src/runtime/handle.rs

```diff
@@ -252,6 +252,15 @@ impl Handle {
     /// timers will not work unless there is another thread currently calling
     /// [`Runtime::block_on`] on the same runtime.
     ///
+    /// If this function is called from within a Tokio runtime's asynchronous
+    /// context, such as inside a [`Runtime::block_on`] scope, or from a function
+    /// annotated with the [`tokio::main`] attribute macro, it will panic, because
+    /// nesting runtimes is not allowed.
+    /// The same applies when calling `Handle::block_on` to re-enter the current
+    /// runtime. To safely run asynchronous code from such a context, you can use
+    /// [`task::block_in_place`] (only available with the multi-thread scheduler),
+    /// as shown in the example below.
```

Comment:
It would be better moving it into the "Panics" section, and then rewrite the current "Panics" section to make shorter.

For example:

```rust
/// # Panics
///
/// This function will panic if any of the following conditions are met:
///
/// 1. The provided future panics.
/// 2. This function is called inside the [`Runtime::block_on`] or [`Handle::block_on`].
```